File: 18/sol1.py
from itertools import chain
from common import tick
from common import read_input


if __name__ == '__main__':
    trees, lumber, clear = read_input()
    maxx = max(c[0] for c in chain(trees, lumber, clear)) + 1
    maxy = max(c[1] for c in chain(trees, lumber, clear)) + 1
    print(maxx, maxy)
    for minute in range(10):
        trees, lumber, clear = tick(trees, lumber, clear, maxx, maxy)
    res = len(trees) * len(lumber)
    print(f'{len(trees)} * {len(lumber)} = {res}')

File: 18/sol1.rb
require './common.rb'

blocks = read_input
all_blocks = blocks.trees | blocks.lumber | blocks.clear
maxx = all_blocks.max_by { |b| b.x }.x + 1
maxy = all_blocks.max_by { |b| b.y }.y + 1
puts("Maxx: #{maxx}, Maxy: #{maxy}")

# Simulate 10 minutes on the initial forest and print out the result
10.times do
  blocks = tick(blocks, maxx, maxy)
end
res = blocks.trees.size * blocks.lumber.size
puts("#{blocks.trees.size} * #{blocks.lumber.size} = #{res}")
